File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/GameChatSystemMessage.java
package tech.zolhungaj.amqapi.servercommands.gameroom;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;

import java.util.Optional;

@CommandType("Game System Chat Message")
public record GameChatSystemMessage(
    String title,
    @Json(name = "msg") Optional<String> message,
    @Json(name = "teamMessage") Boolean isTeamMessage
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/GameChatSystemMessage.kt
package tech.zolhungaj.amqapi.servercommands.gameroom

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType

@JvmRecord
@CommandType("Game System Chat Message")
data class GameChatSystemMessage(
    @Json(name = "message") val message: String
)
